src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -514,12 +515,29 @@ protected function initializeContainer()
         $cache = new ConfigCache($this->getCacheDir().'/'.$class.'.php', $this->debug);
         $fresh = true;
         if (!$cache->isFresh()) {
-            $container = $this->buildContainer();
+            if ($this->debug) {
+                $collectedLogs = array();
+                $previousHandler = set_error_handler(function () use (&$collectedLogs, &$previousHandler) {
+                    $collectedLogs[] = $args = func_get_args();
```

Comment:
this contains the $context argument, which can be anything so potentially unserializable
let's use regular args: $type, $message, $file, $line (and ignore the last $context one)